File: app/src/main/java/com/lucas/omnia/fragments/MyPostsTabFragment.java
package com.lucas.omnia.fragments;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;

public class MyPostsTabFragment extends PostListFragment {

    public MyPostsTabFragment() {
        // Required empty public constructor
    }

    @Override
    public Query getQuery(DatabaseReference databaseReference) {
        // User last 100 posts
        return databaseReference.child("user-posts").child(getUid()).limitToFirst(100);
    }
}

File: app/src/main/java/com/lucas/omnia/fragments/MyPostsTabFragment.kt
package com.lucas.omnia.fragments

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.Query

class MyPostsTabFragment : PostListFragment() {
    override fun getQuery(databaseReference: DatabaseReference?): Query {
        // User last 100 posts
        return databaseReference!!.child("user-posts").child(uid).limitToFirst(100)
    }
}